internal/exec/util/blkid.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package util
+
+// #cgo LDFLAGS: -lblkid
+// #include <stdlib.h>
+// #include "blkid.h"
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/coreos/ignition/config/types"
+)
+
+func FilesystemType(device types.Path) (string, error) {
+	var fsType [16]C.char
+
+	cDevice := C.CString(string(device))
+	defer C.free(unsafe.Pointer(cDevice))
+
+	switch C.filesystem_type(cDevice, (*C.char)(&fsType[0]), C.size_t(len(fsType))) {
+	case 0:
+		return C.GoString(&fsType[0]), nil
+	case -1:
```

Comment:
maybe enum the error returns in C-land so they have some meaningful names rather than these magic numbers?
You can readily access C enums in cgo, they're just added to the C.$foo namespace: https://github.com/giorgisio/cgo/blob/master/enums.go